lib/registration.pm

```diff
@@ -119,12 +135,20 @@ sub fill_in_registration_data {
                     save_screenshot;
                 }
             }
-            send_key 'alt-n';    # next
-            while (check_screen('import-untrusted-gpg-key', 20)) {
-                record_soft_failure 'untrusted gpg key';
-                send_key 'alt-t', 2;
+            send_key 'alt-n';
+            # start addons/modules registration, it needs longer time if select multiple or all addons/modules
+            while (assert_screen(['import-untrusted-gpg-key', 'yast_scc-pkgtoinstall', 'inst-addon'], 120)) {
```

Comment:
because assert_screen returns the matched needle on success and throws an exception on no match. I think the while is misleading here because it will always evaluate to true (any matched needle is true) or has no effect in case of no match when an exception is thrown. It reminds me of `send_key 'ret', 3;`. While syntactically it is correct, lazy people will copy-paste without understanding and IMHO making it worse.